### PR TITLE
Update openssl to 0.9.10

### PR DESCRIPTION
**Pull request by Manishearth**
r? @aatxe

cc @sfackler

### CODE REVIEW

File: src/client/conn.rs [hunk 1]
```diff
@@ -44,70 +45,63 @@ pub trait Connection {
...
     /// Connects over SSL to the specified server and returns a reader-writer pair.
     #[cfg(feature = "ssl")]
-    fn connect_ssl_internal(host: &str, port: u16) -> Result<NetReadWritePair> {
-        let socket = try!(TcpStream::connect(&format!("{}:{}", host, port)[..]));
-        let ssl = try!(ssl_to_io(SslContext::new(SslMethod::Sslv23)));
-        let ssl_socket = try!(ssl_to_io(SslStream::connect_generic(&ssl, socket)));
-        Ok((BufReader::new(NetStream::Ssl(try!(ssl_socket.try_clone()))),
-            BufWriter::new(NetStream::Ssl(ssl_socket))))
+    fn connect_ssl_internal(host: &str, port: u16) -> Result<NetBufStream> {
+        let domain = format!("{}:{}", host, port);
```
Comment (sfackler) [inline src/client/conn.rs]: No need to format here - `(&str, u16)` can be passed to `connect` directly.

File: src/client/conn.rs [hunk 2]
```diff
@@ -44,70 +45,63 @@ pub trait Connection {
...
+        let socket = try!(TcpStream::connect(&domain));
+        let connector = try!(ssl_to_io(SslConnectorBuilder::new(SslMethod::tls()).map(|c| c.build()).map_err(Into::into)));
```
Comment (sfackler) [inline src/client/conn.rs]: There should probably be the option of configuring the connector here - people may want to e.g. add trusted root certificates.

Comment (Manishearth) [inline src/client/conn.rs]: (I only want this updated, so I'm not going to fix this 😄 . Good observation though)

File: src/client/conn.rs (hunk 1 above)
Comment (Manishearth) [inline src/client/conn.rs]: done

**Comment by aatxe** [comment]
Doesn't this change behavior in a multithreaded situation? That is if you have a reading thread, a separate writing thread will wait forever? I believe some situation like this was why I split the reader and writer originally.

cc @retep998 (who yelled at me about this in the first place)

**Comment by Manishearth** [comment]
@sfackler was asking about this on IRC -- if it was duplexed or not. I guess it is?

**Comment by Manishearth** [comment]
We should probably figure something out here soon, currently this library is totally broken if someone wants to use hyper as well (you can't have multiple versions of native libs in your deps)

**Comment by aatxe** [comment]
Agreed. My intention was to handle this while I changed the library to be asynchronous but when I was trying, I couldn't quite figure out a good interface to expose.

**Comment by qlyoung** [comment]
What's the status on this? Still broken with hyper. Love the library but if this patch fixes things perhaps you could take it and patch later?

**Comment by aatxe** [comment]
Well, I suppose I don't have a good sense of how people are using the library, but there's a reasonable chance that this would be a very serious breakage. You won't be able to send messages while waiting to receive a message. You'll only be able to send responses following a received message.

**Comment by aatxe** [comment]
@Manishearth I decided to go ahead with this anyway, but committed it manually with `--author`. I think the duplexing is not likely to be much of an issue for the average user. So, as @qlyoung suggested, we can fix the new problem later.

**Comment by Manishearth** [comment]
Awesome, thanks!

**Pull request closed without merging**